src/gallium/state_trackers/nine/buffer9.c

```diff
@@ -242,8 +242,13 @@ NineBuffer9_Lock( struct NineBuffer9 *This,
                 This->managed.dirty_box = box;
                 if (p_atomic_read(&This->managed.pending_upload))
                     nine_csmt_process(This->base.base.device);
-            } else
-                u_box_union_2d(&This->managed.dirty_box, &This->managed.dirty_box, &box);
+            } else {
```

Comment:
no change needed here, because nine already checks for non-initialized box with managed.dirty.